tests/functional/object/mcg/test_bucket_replication_with_versioning.py

```diff
@@ -0,0 +1,144 @@
+TIMEOUT = 300
+
+
+@mcg
+@red_squad
+@runs_on_provider
+class TestReplicationWithVersioning(MCGTest):
+    """
+    Test suite for MCG object replication policies
+    """
+
+    @pytest.fixture(autouse=True, scope="class")
+    def reduce_replication_delay_setup(self, add_env_vars_to_noobaa_core_class):
+        """
+        Reduce the replication delay to one minute
+
+        Args:
+            new_delay_in_miliseconds (function): A function to add env vars to the noobaa-core pod
+        """
+        new_delay_in_miliseconds = 60 * 1000
+        new_env_var_touples = [
+            (constants.BUCKET_REPLICATOR_DELAY_PARAM, new_delay_in_miliseconds),
+            (constants.BUCKET_LOG_REPLICATOR_DELAY_PARAM, new_delay_in_miliseconds),
+        ]
+        add_env_vars_to_noobaa_core_class(new_env_var_touples)
+
+    @pytest.fixture()
+    def buckets_with_versioning(self, bucket_factory, mcg_obj, awscli_pod_session):
+        """
+        Prepare two buckets with versioning enabled
+
+        Args:
+            bucket_factory: Fixture for creating new buckets
+            mcg_obj_session: The session-scoped MCG object
+            awscli_pod_session: The session-scoped AWSCLI pod
+
+        Returns:
+            Tuple of two buckets: source and target
+        """
+        # Using the OC interface allows patching a repli policy on the OBC
+        bucket_a = bucket_factory(1, "OC")[0]
+        bucket_b = bucket_factory(1, "OC")[0]
+
+        put_bucket_versioning_via_awscli(mcg_obj, awscli_pod_session, bucket_a.name)
+        put_bucket_versioning_via_awscli(mcg_obj, awscli_pod_session, bucket_b.name)
+
+        return bucket_a, bucket_b
+
+    @tier1
+    @polarion_id("OCS-6294")
+    def test_bucket_replication_with_versioning(
+        self,
+        awscli_pod,
+        mcg_obj,
+        buckets_with_versioning,
+    ):
+        """
+        1. Create two buckets and enable versioning on both
+        2. Set a bucket replication policy with versioning enabled on the source bucket
+        3. Write some versions to the source bucket
+        4. Verify the versions were replicated to the target bucket in the same order
+        """
+        obj_key = "test_obj_" + str(uuid4())[:4]
+        versions_amount = 5
+
+        # 1. Create two buckets and enable versioning on both
+        source_bucket, target_bucket = buckets_with_versioning
+
+        # 2. Set a bucket replication policy with versioning enabled on the source bucket
+        replication_policy = ReplicationPolicyWithVersioning(
+            target_bucket=target_bucket.name
+        )
+        update_replication_policy(source_bucket.name, replication_policy.to_dict())
+
+        # 3. Write some versions to the source bucket
+        source_etags = upload_obj_versions(
+            mcg_obj,
+            awscli_pod,
+            source_bucket.name,
+            obj_key=obj_key,
+            amount=versions_amount,
+        )
+        logger.info(f"Uploaded versions: {source_etags}")
```

Comment:
Change string to `Uploaded etags` instead of `Uploaded versions`

Reply:
fixed in https://github.com/red-hat-storage/ocs-ci/pull/10957/commits/41d24f028d85cae4c730f93bfaa79a9ffb210c39